### PR TITLE
Upgrade dependencies to the latest versions

### LINKED ISSUE #559: Ensure that all library dependencies are at latest stable version

### PR DESCRIPTION
**Pull request by mraible**
Remove JUnit references and replace with TestNG. Fix Groovy issues after upgrading to latest version.

Fixes #559.

### CODE REVIEW

File: examples/quickstart/pom.xml [hunk 1]
```diff
@@ -87,11 +87,6 @@
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-compiler-plugin</artifactId>
```
Comment (dogeared) [inline examples/quickstart/pom.xml]: Why remove this? Is it unnecessary?

File: examples/servlet/pom.xml [hunk 2]
```diff
@@ -34,9 +34,9 @@
     <properties>
         <jdk.version>1.6</jdk.version>
         <jstl.version>1.2</jstl.version>
-        <logback.version>1.1.2</logback.version>
+        <logback.version>1.1.7</logback.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <servlet.version>3.0.1</servlet.version>
+        <servlet.version>3.1.0</servlet.version>
```
Comment (dogeared) [inline examples/servlet/pom.xml]: Hooray!

File: examples/quickstart/pom.xml (hunk 1 above)
Comment (mraible) [inline examples/quickstart/pom.xml]: It's a duplicate. It's already defined on line 56 in this file. https://github.com/stormpath/stormpath-sdk-java/blob/887051575aa31c3922192766451e9394bbe19cee/examples/quickstart/pom.xml#L56

Comment (dogeared) [inline examples/quickstart/pom.xml]: gotcha. Thanks.

File: extensions/spring/stormpath-spring-security/pom.xml [hunk 3]
```diff
@@ -64,8 +64,10 @@
             <groupId>org.springframework</groupId>
             <artifactId>spring-test</artifactId>
         </dependency>
-
+        <dependency>
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security/pom.xml]: per Les' concern the other day, does this have any impact on end-developers? Would they need to have hamcrest themselves to use Spring Security?

Comment (mraible) [inline extensions/spring/stormpath-spring-security/pom.xml]: No, it's just for compiling our tests. Apparently it used to be included in one of the dependencies (maybe TestNG?) and now it isn't.

File: pom.xml [hunk 4]
```diff
@@ -306,6 +295,16 @@
                 <version>${jackson.version}</version>
             </dependency>
             <dependency>
+                <groupId>com.fasterxml.jackson.core</groupId>
```
Comment (dogeared) [inline pom.xml]: Is this for servlet? Spring/Spring Boot already has this built-in, right? Is there any impact to Spring and Spring Boot by including it here? For instance, what if we have a later version of this library specified here than Spring uses?

**Comment by dogeared** [comment]
How is the update of `OauthHttpServletRequest` in this PR related to updating the versions?

... back to you @mraible

File: pom.xml (hunk 4 above)
Comment (mraible) [inline pom.xml]: This is added in dependencyManagement so the versions are locked down for any projects that include it as a dependency. You'll notice I had to include `jackson-annotations` and `jackson-core` below. Without this, it was pulling in older versions of those dependencies. This section does not specifically include Jackson as a dependency.

**Comment by mraible** [comment]
@dogeared New methods were added to `HttpServletRequest` in Servlet 3.1, so they needed to be implemented in `OAuthHttpServletRequest`.

File: examples/servlet/pom.xml (hunk 2 above)
Comment (jarias) [inline examples/servlet/pom.xml]: So this means no more Java 6?

Comment (dogeared) [inline examples/servlet/pom.xml]: Yup. We already took that out of what's merged into master.

File: extensions/spring/stormpath-spring-security/pom.xml (hunk 3 above)
Comment (dogeared) [inline extensions/spring/stormpath-spring-security/pom.xml]: 👍

File: pom.xml (hunk 4 above)
Comment (dogeared) [inline pom.xml]: got it. Thanks.

**Comment by dogeared** [comment]
LGTM! Please merge.

**Pull request merged**